bucket/drivers/s3/s3.go

```diff
@@ -0,0 +1,151 @@
+package s3
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+
+	"github.com/sirupsen/logrus"
+)
+
+type S3Driver struct {
+	config *aws.Config
+}
+
+func New(config *aws.Config) (*S3Driver, error) {
+	if config == nil {
+		return nil, fmt.Errorf("must provide S3 config")
+	}
+
+	return &S3Driver{
+		config: config,
+	}, nil
+}
+
+// Returns a new S3 service client
+func (d *S3Driver) NewSvc() (*s3.S3, error) {
+	// Override the aws config with the region
+	sess, err := session.NewSession(d.config)
+	if err != nil {
+		return nil, err
+	}
+
+	svc := s3.New(sess)
+	return svc, nil
+}
+
+func (d *S3Driver) updateRegion(region string) {
+	d.config.Region = aws.String(region)
+}
+
+func (d *S3Driver) CreateBucket(name string, region string) (string, error) {
+	// Update driver region config
+	d.updateRegion(region)
+	svc, err := d.NewSvc()
```

Comment:
Let's just pass in the region to d.NewSvc() instead of updating the global config object. If multiple bucket requests come in at the same time, this might cause a concurrency issue. It is also less reliable if multiple requests come in - regions could be overwritten.